### PR TITLE
Reduce unnecessary render() calls by ensuring ref equality with style objects

### PR DESCRIPTION
**Pull request by bvaughn**
Continuation of PR #483.

### CODE REVIEW

File: source/Grid/Grid.js [hunk 1]
```diff
@@ -701,8 +704,22 @@ export default class Grid extends Component {
   _enablePointerEventsAfterDelayCallback () {
```
Comment (zackargyle) [inline source/Grid/Grid.js]: Might be worth renaming this function now.

**Comment by zackargyle** [comment]
Looks awesome. Curious if using an LRU cache would perform better. You wouldn't have to worry about clearing the cache on scroll end and you'd always have the viewable cell styles in cache. Either way, solid optimization!

**Comment by MalucoMarinero** [comment]
Sorry, haven't lost interest, more like having a week off looking after the family, but either way you guys are probably better situated to make your decisions on how to move this forward, you don't need/want any input from me yeah?

**Comment by bvaughn** [comment]
Input is valuable, definitely. I'd welcome any you have! But family stuff is more important than coding so no worries at all.

File: source/Grid/Grid.js (hunk 1 above)
Comment (bvaughn) [inline source/Grid/Grid.js]: Agreed!

**Comment by bvaughn** [comment]
FYI this optimization has been published as version 8.8.0

**Comment by romulof** [comment]
I think this has broken my code.

I have List with variable-height elements. At start it renders a small "loading" element, then it renders the actual element. Even manually controlling CellMeasurer cache, it still renders the actual element with the same small height as the "loading" element.

**Comment by bvaughn** [comment]
Size info has always been cached. If size changes/invalidates, you need to inform Grid via the public API method telling it that a given cell's size is invalid.

I can't diagnose a brief description of a bug like this though. You should post to Stack Overflow.

**Pull request merged**